packages/theme-default/src/layout/NotFountLayout/index.tsx

```diff
@@ -1,21 +1,43 @@
-import { usePageData } from '@rspress/runtime';
+import { useMemo } from 'react';
+import { usePageData, withBase } from '@rspress/runtime';
 
 export function NotFoundLayout() {
   const { siteData } = usePageData();
   const defaultLang = siteData.lang;
+  const isBrowser = typeof window !== 'undefined';
   // Consider the existing sites include the defaultLang in some links, such as '/zh/guide/quick-start'
   // We need to redirect them to '/guide/quick-start'
   // In the meanwhile, we will not show the 404 page for the user experience
   if (
+    isBrowser &&
     defaultLang &&
-    typeof window !== 'undefined' &&
     location.pathname.includes(`/${defaultLang}/`)
   ) {
     const redirectUrl = location.pathname.replace(`/${defaultLang}/`, '/');
     window.location.replace(redirectUrl);
     return <></>;
   }
 
+  const root = useMemo(() => {
+    if (isBrowser) {
+      const currentLang = location.pathname
```

Comment:
when we enable multiversion, the url will be like `/v2/zh/index.html`, this logic have little problem?

May be we can use `usePageData()` hook, to get the `page.lang` for currentLang? I'm not sure if this will work. You can check to see if there is any relevant implementation of getting the current language in the entire repo.

Reply:
Both of `page.lang` and `useLang` can't work. Maybe I should use the logic above `includes` or make more judgments on version in the changed code. What's your advice?

Reply:
https://github.com/web-infra-dev/rspress/blob/faf6bcf8aae5394db395773da27fb433b33d0b0a/packages/theme-default/src/layout/Layout/index.tsx#L119

pass the currentLang in props here?

Reply:
Should we consider the current version? Go back to the home of the current version in the current language or just the default version in the current language?

Reply:
the current version in the current language is better

Reply:
> pass the currentLang in props here?

This doesn't work. I already have some ideas, and I will open a new PR to solve it.